__tests__/app.test.js

```diff
@@ -1,34 +1,138 @@
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: ARTICLE_ID,
+          title: "Eight pug gifs that remind me of mitch",
+          topic: "mitch",
+          author: "icellusedkars",
+          body: "some gifs",
+          created_at: "2020-11-03T09:12:00.000Z",
+          votes: 0,
+        });
+      });
   });
-});
\ No newline at end of file
+  test("status: 400, responds with a invalid input", () => {
+    const ARTICLE_ID = "not a number";
+    return request(app)
+      .get(`/api/articles/${ARTICLE_ID}`)
+      .expect(400)
+      .then(({ body }) => {
+        const { message } = body;
+        expect(message).toEqual("Invalid data type");
+      });
+  });
+  test("status:404, article not found", () => {
+    const ARTICLE_ID = 1000;
+    return request(app)
+      .get(`/api/articles/${ARTICLE_ID}`)
+      .expect(404)
+      .then(({ body }) => {
+        const { message } = body;
+        expect(message).toEqual("Article not found");
+      });
+  });
+});
+
+describe("1. GET/api?users", () => {
+  test("status:200, responds with an array of users objects", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        const { users } = body;
+        expect(users).toBeInstanceOf(Array);
+        expect(users).toHaveLength(4);
+        users.forEach((user) => {
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  const articleUpdate = {
+    votes: 3,
+  };
+  test("status:200, responds with an updated article", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send(articleUpdate)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 103,
+        });
+      });
+  });
+
+   test("status:404, article not found", () => {
+     return request(app)
+       .get("/api/articles/3000")
+       .expect(404)
+       .then(({ body }) => {
+         const { message } = body;
+         expect(message).toEqual("Article not found");
+       });
+   });
+
+    test("status: 400, responds with a invalid input", () => {
+      const ARTICLE_ID = "not a number";
+      return request(app)
+        .get(`/api/articles/${ARTICLE_ID}`)
+        .expect(400)
+        .then(({ body }) => {
+          const { message } = body;
+          expect(message).toEqual("Invalid data type");
+        });
+    });
+});
```

Comment:
Please have a think of the following tests for .patch:
* What if the inc_votes value is 0?
* What if the inc_votes value is not a number (such as a string like "hello")?
* What if the vote object does not include the key `inc_vote` ?